### ISSUE TITLE
Workbenches and tests not bundling all resources

### ISSUE THREAD

**Issue by thecapdan** [BladeRunnerJS/brjs#1632]
I have found some cases where html and xml is not bundled using latest brjs master. This is a regression since the last brjs release. This will need to be addressed before we can release. More details to follow.

**Comment by thecapdan** [comment]
having had no luck finding unique set of steps to reproduce this issue, I resorted to a git bisect which narrowed it down to this commit: https://github.com/BladeRunnerJS/brjs/commit/3dfe0997192455dad4a60eb4ed81727eef2afc33

which is essentially moving from using a LinkedHashMap instead of a LinkedHashSet to store sourceModules.

**Comment by dchambers** [comment]
@thecapdan, is there any update on this issue?

**Comment by thecapdan** [comment]
@dchambers this has been parked but it's pretty critical we look at it. Or we can revert the NPE fix and reopen https://github.com/BladeRunnerJS/brjs/issues/1517.

It's difficult to determine by looking at https://github.com/BladeRunnerJS/brjs/commit/3dfe0997192455dad4a60eb4ed81727eef2afc33 the difference that would cause the regression.
